vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,27 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+type HealthConfig struct {
+	// Test to perform to check container is healthy.
+	// An empty slice means to inherit the default.
+	// Otherwise, Cmd[0] is "NONE" or "CMD".
+	Test strslice.StrSlice `json:",omitempty"`
+
+	// Nil means to inherit.
+	GracePeriod *float64 `json:",omitempty"` // Time after container start to assume failure means still starting.
```

Comment:
Let's make sure this feedback doesn't get lost:

1. These _must_ be `time.Duration`. This is how time types are represented in Go.
2. Use the zero-value to communicate meaning. For example, in case these are the zero-value, use a default value. There are no need for pointers here.

If the issue is serialization, create a `time.Duration`-compatible that exports methods for conversion to `time.Duration`. However, I see no issue in serialization these an nanosecond-integers.